### PR TITLE
feat(wmg): rewire publication filter to use the data in the cube

### PR DESCRIPTION
**Pull request by atarashansky**
## Recoding the Publication filter to use data in the cube

- #5145
- Wiping the older filter code and replacing it with references to the reengineered cube.
- We remove unnecessary collection calls from useManyCollections() and hopefully lose lag on the webpage. 
- Commenting out analytics calls for now (will re-introduce in another PR).

## Notes for Reviewer
- Will be reverting the skipping e2e tests PR after this is merged!

### CODE REVIEW

**Review by carolinechoi — APPROVED** [review]

File: frontend/src/common/queries/wheresMyGene.ts [hunk 1]
```diff
@@ -122,7 +118,7 @@ export const USE_PRIMARY_FILTER_DIMENSIONS = {
 };
 
 export function usePrimaryFilterDimensions(
-  version: 1 | 2 = 1
+  version: 1 | 2 = 2
```
Comment (seve) [inline frontend/src/common/queries/wheresMyGene.ts:121]: keep default to 2

Comment (seve) [inline frontend/src/common/queries/wheresMyGene.ts:121]: keep default to 2

File: frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveExport/csvUtils.ts [hunk 2]
```diff
@@ -110,17 +102,17 @@ export function csvHeaders({
...
+  // output.push([
+  //   `# Publication Filter Values: ${
+  //     publicationFilter
+  //       ?.filter((option) => {
+  //         return selectedPublicationFilter.publications.includes(option.id);
+  //       })
+  //       .map((selected) => selected.name)
+  //       .join(", ") || NO_SELECTION_STRING
+  //   }`,
+  // ]);
```
Comment (seve) [inline frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveExport/csvUtils.ts:115]: does this need to be deleted

File: frontend/tests/utils/downloadUtils.ts [hunk 3]
```diff
@@ -233,9 +233,9 @@ const verifyFilterValues = (
     case "self-reported-ethnicity-filter":
       ethnicityFilterValue = filterText || noSelectionText;
       break;
-    case "publication-filter":
-      publicationFilterValue = filterText || noSelectionText;
-      break;
+    // case "publication-filter":
+    //   publicationFilterValue = filterText || noSelectionText;
+    //   break;
```
Comment (seve) [inline frontend/tests/utils/downloadUtils.ts:238]: should this be uncommented?

File: frontend/tests/utils/downloadUtils.ts [hunk 4]
```diff
@@ -256,9 +256,9 @@ const verifyFilterValues = (
       `# Self-Reported Ethnicity Filter Values: ${ethnicityFilterValue}`
     )
   ).toBeTruthy();
-  expect(
-    data[6].includes(`# Publication Filter Values: ${publicationFilterValue}`)
-  ).toBeTruthy();
+  // expect(
+  //   data[6].includes(`# Publication Filter Values: ${publicationFilterValue}`)
+  // ).toBeTruthy();
```
Comment (seve) [inline frontend/tests/utils/downloadUtils.ts:261]: uncommented?

**Review by seve — CHANGES_REQUESTED** [review]
can you verify tissue filter and share link work?

**Pull request merged**